docs/use_cases/retrieval_augmented_generation_eval.md

```diff
@@ -0,0 +1,181 @@
+
+A [utility](https://chunkviz.up.railway.app/) like this seems useful to visualise your apparent chunks.
+
+### Semantic Retrieval Evaluation
+
+The next step is semantic retrieval evaluation. It puts the work you've done so far to a litmus test.
+
+Retrieval is a vital component of RAG and its evaluation can be addressed as a classic information retrieval evaluation problem.
+
+You need to establish the expectations from the returned results. This can help
+us to identify reference metrics and important parameters. Done correctly, it
+helps us determine if the documents retrieved at this stage are relevant
+results.
+
+We have several existing metrics to guide and define our baseline:
+
+- Precision and Recall or their combination F1 Score
+- DCG and nDCG, which relates to their relevance, based on the inclusion or rank of documents in the results
+
+The nature of semantic information retrieval poses a challenge at this stage, as the documents are retrieved beyond the keywords/synonyms/token enrichment-matching.
+
+You can build a reference evaluation set, known as a [Golden Set](https://www.luigisbox.com/search-glossary/golden-set/). We could also leverage [T5 Model](https://huggingface.co/docs/transformers/model_doc/t5) to generate a starter pack for evaluation.
+
+The golden set is a fundamental component in information retrieval evaluation.
+It helps in define characteristics of a reference standard to assess the performance, effectiveness, and relevance of a selected retrieval algorithm.
+It provides a common ground for objective comparison and improvement of a retrieval process/algorithm.
+
+
+### End-to-End Evaluation
+
+An end-to-end evaluation covers the response generation of the question. It leverages the provided context through retrieved documents.
+
+Evaluating the quality of responses produced by large language models can be a challenge due to various factors as described above. 
+
+By virtue of nature and design, the answers generated rely on the diversity of response which makes it impossible to devise a fixed metric or methodology that fits all domains and use-cases.
+
+To address these difficulties, you may use a blend of existing metrics like the following scores: 
+
+- [BLEU](https://huggingface.co/spaces/evaluate-metric/bleu) 
+- [ROUGE](https://huggingface.co/spaces/evaluate-metric/rouge) 
+
+You can then combine these score with LLM-based or human evaluation methods.
+For more information, see this paper which provides great ideas to establish [Quality Criteria](https://scholarspace.manoa.hawaii.edu/server/api/core/bitstreams/c6a53998-09e3-4d17-91fd-c7416d51b250/content) for the same.
+
+To summarize, you want to establish methods to automate evaluating similarity
+and content overlap between generated response and reference summaries. You can
+also leverage human evaluation to evaluate subjective information, such as context-relevance, novelty, and fluency.
```

Comment:
ahh, it is explained here, I would somehow rework this paragraph and the previous one together